File: maplay/Data/Entities/Spot.cs
using NetTopologySuite.Geometries;

namespace Maplay.Data.Entities;

public class Spot
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Status { get; set; } = SpotStatus.Pending;
    public string SpotType { get; set; } = SpotTypes.Permanent;
    public DateOnly? StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public string[] AgeGroups { get; set; } = Array.Empty<string>();
    public string[] Facilities { get; set; } = Array.Empty<string>();
    public Point Location { get; set; } = default!;
    public string? Address { get; set; }
    public Guid? SubmittedBy { get; set; }
    public string Source { get; set; } = SpotSources.User;
    public string? GovDataId { get; set; }
    public Guid? ReviewedBy { get; set; }
    public DateTime? ReviewedAt { get; set; }
    public string? RejectReason { get; set; }
    public string? SourceUrl { get; set; }
    public DateTime? CrawledAt { get; set; }
    public DateTime? DeletedAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<SpotImage> Images { get; set; } = new List<SpotImage>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}

public static class SpotStatus
{
    public const string Pending = "pending";
    public const string Approved = "approved";
    public const string Rejected = "rejected";
}

public static class SpotTypes
{
    public const string Permanent = "permanent";
    public const string Temporary = "temporary";
}

public static class SpotSources
{
    public const string User = "user";
    public const string GovImport = "gov_import";
}

public static class SpotCategories
{
    public static readonly HashSet<string> All = new()
    {
        "park", "restaurant", "nursing_room", "toilet", "activity", "medical"
    };
}

public static class AgeGroupValues
{
    public static readonly HashSet<string> All = new() { "0-3", "3-7", "7-12", "12+" };
}

public static class FacilityValues
{
    public static readonly HashSet<string> All = new()
    {
        "diaper_table", "nursing_room", "stroller_accessible", "kids_toilet",
        "high_chair", "parking", "elevator", "play_area"
    };
}

File: frontend/src/types/spot.ts
// Spot related types
export interface Spot {
  id: number
  name: string
  category: SpotCategory
  spotType: 'permanent' | 'temporary'
  address: string
  lat: number
  lng: number
  description?: string
  ageGroups: AgeGroup[]
  facilities: Facility[]
  startDate?: string // ISO 8601 date string
  endDate?: string // ISO 8601 date string
  status: 'pending' | 'approved' | 'rejected'
  submittedBy?: number
  createdAt: string // ISO 8601 datetime string
  updatedAt: string // ISO 8601 datetime string
  deletedAt?: string // ISO 8601 datetime string
  distanceMeters?: number // For nearby queries
  images?: SpotImage[]
  averageRating?: number
  averageCleanLevel?: number
  reviewCount?: number
  reviews?: Review[]
}

export interface SpotImage {
  id: number
  spotId: number
  url: string
  isCover: boolean
  uploadedBy: number
  createdAt: string
}

export interface Review {
  id: number
  spotId: number
  userId: number
  rating: number
  cleanLevel: number
  comment?: string
  createdAt: string
  updatedAt: string
}

export type SpotCategory =
  | 'park'
  | 'restaurant'
  | 'nursing-room'
  | 'medical'
  | 'activity'
  | 'other'

export type AgeGroup = '0-3' | '3-7' | '7-12' | '12+'

export type Facility =
  | 'parking'
  | 'stroller-accessible'
  | 'diaper-changing-station'
  | 'playground'
  | 'rest-area'
  | 'feeding-room'
  | 'toilet'
  | 'water-fountain'
  | 'shaded-area'
  | 'wifi'
  | 'other'

// API Request/Response types
export interface NearbySpotsQuery {
  lat: number
  lng: number
  radius?: number
  category?: SpotCategory
  age?: AgeGroup
  spotType?: 'permanent' | 'temporary'
  page?: number
  pageSize?: number
}

export interface CreateSpotRequest {
  name: string
  category: SpotCategory
  spotType: 'permanent' | 'temporary'
  address: string
  lat: number
  lng: number
  description?: string
  ageGroups: AgeGroup[]
  facilities: Facility[]
  startDate?: string
  endDate?: string
}

export interface NearbySpotsResponse {
  data: Spot[]
  total: number
  page: number
  pageSize: number
}